Core.Auditing/IfxAuditLogger.cs

```diff
@@ -1,101 +1,124 @@
+            foreach (CallerIdentity callerIdentity in callerIdentities)
             {
-                /* In local box environment, audit logging mightn't be saved.
-                In order to locally debug your code, please download and execute RunBefore_IFxAuditWebAppsLocalTesting.cmd, 
-                this sets a couple of environment variables that are present in Web Apps and IFxAudit needs.After running that please make sure you restart
-                Visual Studio so that the changes are picked up.(See https://genevamondocs.azurewebsites.net/collect/references/ifxref/ifxAuditWebApps.html)
-                */
-                Dictionary<string, string> properties = new Dictionary<string, string>()
-                {
-                    { "IfxResultMessage", ifxResult.Message },
-                    { "IfxResultCode", ifxResult.Code.ToString() },
-                };
-                telemetryClient.TrackEvent("IfxApplicationAuditFailure", properties);
-                alreadyLoggedPerSession = true;
+                auditRecord.AddCallerIdentity(callerIdentity.type, callerIdentity.name, callerIdentity.description);
             }
-            else
+
+            foreach (TargetResource targetResource in targetResources)
             {
-                // If there is ever success for sending the logs, reset "alreadyLoggedPerSession" so that we capture transient failures more accurately.
-                alreadyLoggedPerSession = false;
+                auditRecord.AddTargetResource(targetResource.type, targetResource.name);
+            }
+
+            auditRecord.OperationAccessLevel = "GroupManager";
+            auditRecord.CallerIpAddress = this.ipAddress;
+            auditRecord.AddCallerAccessLevel("GroupOperator"); 
+            auditRecord.CallerAgent = this.webAppName;
+
+            try
+            {
+                this.logger.LogAudit(auditRecord);
+            }
+            catch (Exception ex)
+            {
+                telemetryClient.TrackException(ex, "GenevaAuditFailure");
             }
         }
 
-        public void LogTokenGenerationAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, TargetResource[] targetResources, CallerIdentity[] callerIdentities, string tokenType)
+        public void LogTokenGenerationAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, string operationResultDescription, List<TargetResource> targetResources, List<CallerIdentity> callerIdentities, string tokenType)
         {
-            LogAuditEvent(telemetryClient,  targetResources, callerIdentities, tokenType + TokenGenerationOperation, operationResult, AuditEventCategory.Authorization);
+            LogAuditEvent(telemetryClient, OperationCategory.Authorization, OperationType.Read, tokenType + TokenGenerationOperation, operationResult, operationResultDescription, callerIdentities, targetResources);
```

Comment:
Is TokenGeneration a Read operation?

Reply:
The options are : 

Read
Update
Create
Delete

You think that Create would be more appropriate ?

Reply:
Yes. If creating an access token is what we should log, and not just using it, then _Create_ seems like the right choice to me.